packages/web/src/components/AddNewShelfForm/AddNewShelfForm.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, FormEvent, useEffect } from 'react';
+import {
+  Form,
+  SubmitButton,
+} from 'elements';
+import './styles.scss';
+import useCustomForm from '../../hooks/useCustomForm';
+
+type Props = {
+  onSubmit: (name: string, location: string, description: string) => void;
+};
+
+interface FormFields {
+  name?: string;
+  location?: string;
+  description?: string;
+}
+
+export const AddNewShelfForm = ({ onSubmit }: Props) => {
+  const initialValues = {
+    name: '',
+    location: '',
+    description: '',
+  };
+
+  const nameErrorMessage = (name: string): string | null => {
+    if (name.length < 3 || name.length > 15) {
```

Comment:
"My Awesome Shelf" will not fit max length. I'd suggest change it to common 256 symbols

Reply:
Fixed as agreed with @belgithub